treescript/src/treescript/script.py

```diff
@@ -37,23 +67,44 @@
             num_changes += await bump_version(config, task, repo_path)
         elif "l10n_bump" == action:
             num_changes += await l10n_bump(config, task, repo_path)
+        elif MERGE_ACTION_WORD == action:
+            pass
         elif "push" == action:
             pass  # handled after log_outgoing
         else:
             raise NotImplementedError("Unexpected action")
+
     num_outgoing = await log_outgoing(config, task, repo_path)
     if num_outgoing != num_changes:
         raise TreeScriptError("Outgoing changesets don't match number of expected changesets!" " {} vs {}".format(num_outgoing, num_changes))
+
+    if should_push(task, actions, num_changes):
+        await push(config, task, repo_path)
+
+    await strip_outgoing(config, task, repo_path)
+
+
+def should_push(task, actions, num_changes):
+    """Decide whether we should push based on task spec.
+
+    Args:
+        task (dict): The running task
+        actions (list): The actions we must perform
+        num_changes (int): The number of commits made, excluding merges.
+
+    Returns:
+        bool
+    """
     if is_dry_run(task):
         log.info("Not pushing changes, dry_run was forced")
     elif "push" in actions:
-        if num_changes:
-            await push(config, task, repo_path)
+        if num_changes or MERGE_ACTION_WORD in actions:
```

Comment:
I think the handling of `num_changes` should be handled outside this function. I've done this refactoring in #129.

Reply:
I'm not clear on what you mean, here. If it's "Use should_push instead" then great. If not then the linked PR looks the same as the original line in this part of the diff

Reply:
The relevant difference to what you implemented here was that `should_push` there doesn't look at `num_changes` (that is now handled in the caller), and so `should_push` does not need to special case the  `merge` action.

So, you should be able to use `should_push` from that PR as-is.